### ISSUE TITLE
[feature request] split `algod`, `kmd`, and `indexer` into features

### ISSUE THREAD

**Issue by skylerqpq** [manuelmauro/algonaut#125]
**Is your feature request related to a problem? Please describe.**
I use `algonaut` in a project, but specifically only use `algod`, I would like to not need to compile `kmd` and `indexer` related code. This isn't a huge deal for the most part, it's just that all code has to be compiled.

**Describe the solution you'd like**
add the following features to the `algonaut_model`, `algonaut_client` and `algonaut`: `"kmd"`, `"indexer"`, `"algod"`, (and add them to default features if that is desirable, presumably it is).

`#[cfg(feature = "<appropriate feature>")]` for `kmd`, `indexer`, and `algod` modules across the aforementioned crates

Sadly this is a breaking change (for `default-features = false` users at the least) so it needs to be accompanied by a minor version bump (0.x -> 0.y)

**Describe alternatives you've considered**
1. Keep with the status quo.
2. Only add some of those `feature`s but not others
3. Add `no-*` features. This is considered non-idiomatic, but has the advantage of being non-breaking.

**Additional context**
N/A

**Comment by ivanschuetz** [comment]
Interesting. I'd estimate that most people don't use at least kmd, so being able to leave services out for sure has a point. A more radical (but cleaner IMO) approach would be to put them in separate projects.

But also, how justified is this? Given incremental compilation and that these services are pretty lean code wise it seems a rather small optimization. The macro resolution might be a bit costly, but still?

**Comment by AlterionX** [comment]
It's has been a bit but agreed on the cleaner version, where the modules get split up based on each service instead of by model/client/etc. My reasoning being that the algod client can change independently of indexer & kmd.

The justification is kinda dependent on the context by which this is used. Technically, if it's delivered often enough, even 1% could be significant and `algonaut`'s a library, so... That said, it'd probably be better to at least check the compilation sizes before saying whether it should or shouldn't happen. Might be more effective looking into how well we can trim down dependencies via flags.

**Comment by ivanschuetz** [comment]
Yeah, the binary sizes (and possibly compilation times) would help us prioritizing this.

Noting that we've the [Algorand test suite](https://github.com/algorand/algorand-sdk-testing), which depends on all 3 projects (I personally question the need for this, but we can't realistically change them), and which we are executing as part of the Github CI. I'm not sure about the exact flow, but it should continue to be executed with the CI of all the projects.

**Comment by manuelmauro** [comment]
This solution would be ideal also in case we would like to add some higher level features to algonaut. In particular, I was thinking of crates for [ARC3 (NFTs)](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0003.md) or [DID](https://github.com/algorandfoundation/did-algo).

**Comment by manuelmauro** [comment]
Resolved more thoroughly than the original request: algod, kmd, and indexer are now three separate crates — [`algonaut_algod`](https://crates.io/crates/algonaut_algod), [`algonaut_kmd`](https://crates.io/crates/algonaut_kmd), [`algonaut_indexer`](https://crates.io/crates/algonaut_indexer) (all at 0.5.0). You can depend on just the one you need without pulling in the others.
